### PR TITLE
docs: refactor kubernetes discovery

### PR DESCRIPTION
**Pull request by zhixiongdu027**
### Description

This Is doc for
#7569 
#7895

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by zhixiongdu027** [comment]
@spacewander @tzssangglass @tokers 
Hi, docs is ready for review , Please help . 
TKS,  @_@

File: docs/en/latest/discovery/kubernetes.md [hunk 1]
```diff
@@ -24,12 +24,17 @@ title: Kubernetes
...
-Discovery also provides a query interface in accordance with the [_APISIX Discovery Specification_](https://github.com/apache/apisix/blob/master/docs/en/latest/discovery.md)
+then store theirs value into ngx.shared.DICT \
+Discovery also provides a node query interface in accordance with the [_APISIX Discovery Specification_](https://github.com/apache/apisix/blob/master/docs/en/latest/discovery.md)
 
-## Configuration
+## How To Use
 
-A detailed configuration for the kubernetes service discovery is as follows:
+kubernetes service discovery both support single-cluster and multi-cluster mode, \
+applicable to the case where the service is distributed in a single or multiple kubernetes clusters.
```
Comment (tzssangglass) [inline docs/en/latest/discovery/kubernetes.md:33]:
```suggestion
Kubernetes service discovery both support single-cluster and multi-cluster mode, applicable to the case where the service is distributed in single or multiple Kubernetes clusters.
```

Comment (tzssangglass) [reply docs/en/latest/discovery/kubernetes.md:33]: `kubernetes` --> `Kubernetes` in this article, I remember that this is the proprietary name.

File: conf/config-default.yaml [hunk 2]
```diff
@@ -303,6 +302,10 @@ nginx_config:                     # config for render the template to generate n
 #       path: "logs/consul_kv.dump"
 #       expire: 2592000            # unit sec, here is 30 day
 #  kubernetes:
+#    ### kubernetes service discovery both support single-cluster and multi-cluster mode
+#    ### applicable to the case where the service is distributed in a single or multiple kubernetes clusters.
+#
+#    ### Single-Cluster Mode ###
```
Comment (SylviaBABY) [inline conf/config-default.yaml:308]: Hi, I see `single-cluster` appearing multiple times in this article. Please keep the same way of writing, either `single-cluster` or `Single-Cluster`.

File: docs/zh/latest/discovery/kubernetes.md [hunk 3]
```diff
@@ -23,13 +23,17 @@ title: Kubernetes
...
-模块同时遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供了节点查询接口
+Kubernetes 服务发现以 [_List-Watch_](https://kubernetes.io/docs/reference/using-api/api-concepts) 方式监听 [_Kubernetes_](https://kubernetes.io) 集群 [_Endpoints_](https://kubernetes.io/docs/concepts/services-networking/service) 资源的实时变化，
+资源，并将其值存储到 ngx.shared.DICT 中 \
+同时遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供了节点查询接口
 
-## Kubernetes 服务发现模块的配置
+## Kubernetes 服务发现的使用
 
-Kubernetes 服务发现模块的完整配置如下：
+Kubernetes 服务发现支持单集群，多集群模式。分别适用于待发现的服务分布在单个或多个 Kubernetes 的情况。
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:32]:
```suggestion
目前 Kubernetes 服务发现支持单集群和多集群模式，分别适用于待发现的服务分布在单个或多个 Kubernetes 的场景。
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 4]
```diff
@@ -73,20 +77,23 @@ discovery:
...
     # for the expression of label_selector, please refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
     label_selector: |-
       first="a",second="b"
+
+    # reserved lua shared memory size, 1m memory can store about 1000 pieces of endpoint
+    shared_size: 1m #default 1m
 ```
 
-如果 Kubernetes 服务发现模块运行在 Pod 内，你可以使用最简配置：
+如果 Kubernetes 服务发现运行在 Pod 内，你可以使用最简配置
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:89]:
```suggestion
如果 Kubernetes 服务发现运行在 Pod 内，你可以使用如下最简配置：
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 5]
```diff
@@ -97,12 +104,12 @@ discovery:
...
       token: # enter serviceaccount token value here
-      #token_file: # enter file path here
+      #token_file: # enter token file path here
 ```
 
-## Kubernetes 服务发现模块的查询接口
+### 单集群模式 Kubernetes 服务发现的查询接口
 
-Kubernetes 服务发现模块遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供查询接口
+单集群模式 Kubernetes 服务发现遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供节点查询接口
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:112]:
```suggestion
单集群模式 Kubernetes 服务发现遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供节点查询接口。
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 6]
```diff
@@ -131,9 +138,121 @@ service_name 必须满足格式: [namespace]/[name]:[portName]
...
+    # kubernetes discovery support label_selector
+    # for the expression of label_selector, please refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
+    label_selector: |-
+      first="a",second="b"
+
+    # reserved lua shared memory size,1m memory can store about 1000 pieces of endpoint
+    shared_size: 1m #default 1m
+```
+
+多集群模式 Kubernetes 服务发现没有为 service,client 域填充默认值，你需要根据集群配置情况自行填充。
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:217]:
```suggestion
多集群模式 Kubernetes 服务发现没有为 `service` 和 `client` 域填充默认值，你需要根据集群配置情况自行填充。
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 7]
```diff
@@ -131,9 +138,121 @@ service_name 必须满足格式: [namespace]/[name]:[portName]
...
+
+### 多集群模式 Kubernetes 服务发现的查询接口
+
+多集群模式 Kubernetes 服务发现遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供节点查询接口
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:221]:
```suggestion
多集群模式 Kubernetes 服务发现遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供节点查询接口。
```

File: docs/en/latest/discovery/kubernetes.md [hunk 8]
```diff
@@ -133,9 +141,123 @@ the kubernetes service discovery provides a query interface in accordance with t
...
+    # reserved lua shared memory size,1m memory can store about 1000 pieces of endpoint
+    shared_size: 1m #default 1m
+```
+
+Multi-Kubernetes service discovery does not fill default values for service and client fields,\
+you need to fill them according to cluster configuration.
+
+### Multi-Cluster Mode Query Interface
+
+the Kubernetes service discovery provides a query interface in accordance with the [_APISIX Discovery Specification_](https://github.com/apache/apisix/blob/master/docs/en/latest/discovery.md)
```
Comment (SylviaBABY) [inline docs/en/latest/discovery/kubernetes.md:225]:
```suggestion
The Kubernetes service discovery provides a query interface in accordance with the [_APISIX Discovery Specification_](https://github.com/apache/apisix/blob/master/docs/en/latest/discovery.md).
```

File: docs/en/latest/discovery/kubernetes.md [hunk 9]
```diff
@@ -133,9 +141,123 @@ the kubernetes service discovery provides a query interface in accordance with t
...
+    # kubernetes discovery support label_selector
+    # for the expression of label_selector, please refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
+    label_selector: |-
+      first="a",second="b"
+
+    # reserved lua shared memory size,1m memory can store about 1000 pieces of endpoint
+    shared_size: 1m #default 1m
+```
+
+Multi-Kubernetes service discovery does not fill default values for service and client fields,\
```
Comment (SylviaBABY) [inline docs/en/latest/discovery/kubernetes.md:220]:
```suggestion
Multi-Kubernetes service discovery does not fill default values for service and client fields, you need to fill them according to the cluster configuration.
```

File: docs/en/latest/discovery/kubernetes.md [hunk 10]
```diff
@@ -101,25 +109,25 @@ discovery:
       #token_file: # enter file path here
 ```
 
-## Interface
+### Single-Cluster Mode Query Interface
 
-the kubernetes service discovery provides a query interface in accordance with the [_APISIX Discovery Specification_](https://github.com/apache/apisix/blob/master/docs/en/latest/discovery.md)
+the Kubernetes service discovery provides a query interface in accordance with the [_APISIX Discovery Specification_](https://github.com/apache/apisix/blob/master/docs/en/latest/discovery.md)
```
Comment (SylviaBABY) [inline docs/en/latest/discovery/kubernetes.md:114]:
```suggestion
The Kubernetes service discovery provides a query interface in accordance with the [_APISIX Discovery Specification_](https://github.com/apache/apisix/blob/master/docs/en/latest/discovery.md).
```

File: docs/en/latest/discovery/kubernetes.md [hunk 11]
```diff
@@ -73,20 +78,23 @@ discovery:
...
-If the kubernetes service discovery runs inside a pod, you can use minimal configuration:
+If the Kubernetes service discovery runs inside a pod, you can use minimal configuration:
 
 ```yaml
 discovery:
   kubernetes: { }
 ```
 
-If the kubernetes service discovery runs outside a pod, you need to create or select a specified [_ServiceAccount_](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/),
+If the Kubernetes service discovery runs outside a pod, you need to create or select a specified [_ServiceAccount_](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/),
```
Comment (SylviaBABY) [inline docs/en/latest/discovery/kubernetes.md:97]:
```suggestion
If the Kubernetes service discovery runs outside a pod, you need to create or select a specified [_ServiceAccount_](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/), then get its token value, and use following configuration:
```

File: docs/en/latest/discovery/kubernetes.md [hunk 12]
```diff
@@ -73,20 +78,23 @@ discovery:
...
 then get its token value, and use following configuration:
```
Comment (SylviaBABY) [inline docs/en/latest/discovery/kubernetes.md:98]:
```suggestion
```

File: docs/en/latest/discovery/kubernetes.md [hunk 13]
```diff
@@ -133,9 +141,123 @@ the kubernetes service discovery provides a query interface in accordance with t
...
+    # for the expression of label_selector, please refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
+    label_selector: |-
+      first="a",second="b"
+
+    # reserved lua shared memory size,1m memory can store about 1000 pieces of endpoint
+    shared_size: 1m #default 1m
+```
+
+Multi-Kubernetes service discovery does not fill default values for service and client fields,\
+you need to fill them according to cluster configuration.
```
Comment (SylviaBABY) [inline docs/en/latest/discovery/kubernetes.md:221]:
```suggestion
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 14]
```diff
@@ -23,13 +23,17 @@ title: Kubernetes
 
 ## 基于 Kubernetes 的服务发现
 
-Kubernetes 服务发现模块以 [_List-Watch_](https://kubernetes.io/docs/reference/using-api/api-concepts) 方式监听 [_Kubernetes_](https://kubernetes.io) 集群 [_Endpoints_](https://kubernetes.io/docs/concepts/services-networking/service) 资源的实时变化，
-并将其值存储到 ngx.shared.kubernetes 中 \
-模块同时遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供了节点查询接口
+Kubernetes 服务发现以 [_List-Watch_](https://kubernetes.io/docs/reference/using-api/api-concepts) 方式监听 [_Kubernetes_](https://kubernetes.io) 集群 [_Endpoints_](https://kubernetes.io/docs/concepts/services-networking/service) 资源的实时变化，
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:26]:
```suggestion
Kubernetes 服务发现以 [_List-Watch_](https://kubernetes.io/docs/reference/using-api/api-concepts) 方式监听 [_Kubernetes_](https://kubernetes.io) 集群 [_Endpoints_](https://kubernetes.io/docs/concepts/services-networking/service) 资源的实时变化，并将其值存储到 ngx.shared.DICT 中。
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 15]
```diff
@@ -23,13 +23,17 @@ title: Kubernetes
...
+资源，并将其值存储到 ngx.shared.DICT 中 \
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:27]:
```suggestion
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 16]
```diff
@@ -23,13 +23,17 @@ title: Kubernetes
...
+同时遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供了节点查询接口
```
Comment (SylviaBABY) [inline docs/zh/latest/discovery/kubernetes.md:28]:
```suggestion

同时遵循 [_APISIX Discovery 规范_](https://github.com/apache/apisix/blob/master/docs/zh/latest/discovery.md) 提供了节点查询接口。
```

**Comment by zhixiongdu027** [comment]
@SylviaBABY 
Tks your review ,very nice.  @_@

**Review by tzssangglass — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Comment by SylviaBABY** [comment]
Since there are some formatting problems in these documents, this PR will be merged first, and I will modify these small errors later.

**Pull request merged**